packages/from-solana/src/SolanaDomainWallet.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  makeSolanaSigningMessageFromRawTransaction,
+  makeSolanaSigningMessageFromStructuredMessage,
+  SolanaDomainIdentity,
+  SolanaDomainPublicKey,
+} from './crypto';
+import { encodeStructuredMessage } from './shared';
+
+const authenticationFunction = '0x7::solana::authenticate';
+
+export interface SolanaDomainWalletOptions {
+  domain?: string;
+  defaultNetwork?: Network;
+}
+
+export class SolanaDomainWallet implements AptosWallet {
+  readonly solanaWallet: SolanaWalletAdapter;
+  readonly domain: string;
+  readonly defaultNetwork: Network;
+
+  readonly version = "1.0.0";
+  readonly name: string;
+  readonly icon: WalletIcon;
+  readonly url: string;
+  readonly accounts = [];
+  readonly chains = APTOS_CHAINS;
+
+  constructor(solanaWallet: SolanaWalletAdapter, options: SolanaDomainWalletOptions = {}) {
+    const { domain, defaultNetwork = Network.MAINNET } = options;
+    this.solanaWallet = solanaWallet;
+    this.domain = domain ?? window.origin;
+    this.defaultNetwork = defaultNetwork;
+    this.name = `${solanaWallet.name} (Solana)`;
+    this.icon = solanaWallet.icon;
+    this.url = solanaWallet.url;
+
+    // const eventsFeature = this.solanaWallet.wallet.features['standard:events'];
+    // console.log(this.solanaWallet.name, 'initial value', this.solanaWallet.wallet.accounts);
+    // eventsFeature.on('change', (args) => {
+    //   console.log('changed', args);
+    // });
+    // this.solanaWallet.on('connect', (account) => {
+    //   console.log(this.solanaWallet.name, 'connected', account);
+    // });
+  }
+
+  readonly features: AptosFeatures = {
+    'aptos:connect': {
+      version: '1.0.0',
+      connect: () => this.connect(),
+    },
+    'aptos:disconnect': {
+      version: '1.0.0',
+      disconnect: () => this.disconnect(),
+    },
+    'aptos:account': {
+      version: '1.0.0',
+      account: () => this.getActiveAccount(),
+    },
+    'aptos:onAccountChange': {
+      version: '1.0.0',
+      // Theoretically supported, but not sure how
+      onAccountChange: async (_callback) => {
+      },
+    },
+    'aptos:network': {
+      version: '1.0.0',
+      network: async () => this.getActiveNetwork(),
+    },
+    'aptos:onNetworkChange': {
+      version: '1.0.0',
+      // Not applicable
+      onNetworkChange: async (_callback) => {
+      },
+    },
+    "aptos:signMessage": {
+      version: '1.0.0',
+      signMessage: (args) => this.signMessage(args),
+    },
+    "aptos:signTransaction": {
+      version: '1.0.0',
+      signTransaction: (...args) => this.signTransaction(...args),
+    },
+  }
+
+  get activeSolanaIdentity(): SolanaDomainIdentity {
+    if (!this.solanaWallet.publicKey) {
+      throw new Error('Account not connected');
+    }
+    return {
+      domain: this.domain,
+      solanaPublicKey: this.solanaWallet.publicKey,
+    };
+  }
+
+  async connect(): Promise<UserResponse<AptosConnectOutput>> {
```

Comment:
this should return the origin wallet account info

Reply:
mm you mean Solana's?
This is the implementation of `aptos:connect` though

Reply:
We need a feature or an exposed API to return the current origin wallet account info (address, public key, etc)

Reply:
let's add this in another PR